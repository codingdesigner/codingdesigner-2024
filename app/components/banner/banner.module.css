.headings {
  composes: headingDefaults from "../typography/typography.module.css";
}

.banner {
  composes: headings;
  align-content: center;
  display: grid;
  gap: var(--spacing-0-5);
  height: auto;
  width: fit-content;

  font-size: var(--type-scale-3);
  line-height: var(--line-height-3);
  transform: translateY(16px);
  overflow: hidden;
}

.banner:hover {
  .bannerWrapper {
    background-color: var(--color-accent-5);
  }
}

.bannerLink {
  color: light-dark(var(--color-grayscale-black), var(--color-grayscale-white));
  text-decoration: none;
}

.bannerLink:hover {
  color: light-dark(var(--color-grayscale-black), var(--color-grayscale-white));
}

.bannerWrapper {
  padding: var(--spacing-0-5);
  display: inline-block;
  position: relative;
}

.bannerItem {
  position: relative;
  display: inline-block;
}

.gwendyWendell {
  composes: bannerLink;
  composes: headings;
  display: block;
}

.gwendyWrapper {
  composes: bannerWrapper;
  background-color: var(--color-accent-1);
  transform: rotate(-1deg) translate3d(-3px, -4px, 0);
}
.gwendy {
  composes: bannerItem;
  transform: rotate(1deg) translate3d(3px, 4px, 0);
}

.wendellWrapper {
  composes: bannerWrapper;
  background-color: var(--color-accent-2);
  transform: rotate(1.5deg) translate3d(3px, 2px, 0);
}
.wendell {
  composes: bannerItem;
  transform: rotate(-1.5deg) translate3d(-3px, -2px, 0);
}

.tcd {
  composes: bannerLink;
  composes: headings;
  display: block;
  transform: translateY(-8px);
}

.the {
  composes: bannerItem;
  font-size: var(--type-scale-1);
  line-height: var(--line-height-1);
}

.codingWrapper {
  composes: bannerWrapper;
  background-color: var(--color-accent-3);
  transform: rotate(0.5deg) translate3d(2px, 2px, 0);
}

.coding {
  composes: bannerItem;
  transform: rotate(-0.5deg) translate3d(-2px, -2px, 0);
}

.designerWrapper {
  composes: bannerWrapper;
  background-color: var(--color-accent-4);
  transform: rotate(-2deg) translate3d(3px, -3px, 0);
}

.designer {
  composes: bannerItem;
  transform: rotate(2deg) translate3d(-3px, 3px, 0);
}

/* only Safari needs this */
@supports not (color: light-dark(red, pink)) {
  .bannerLink {
    color: var(--color-grayscale-black);
  }

  .bannerLink:hover {
    color: var(--color-grayscale-black);
  }
  @media (prefers-color-scheme: dark) {
    .bannerLink {
      color: var(--color-grayscale-white);
    }

    .bannerLink:hover {
      color: var(--color-grayscale-white);
    }
  }
}
