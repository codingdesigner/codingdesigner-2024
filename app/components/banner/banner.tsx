import React from 'react';

import styles from './banner.module.css';

export function Banner() {
  return (
    <div role='banner' className={styles.banner}>
      <a href='/' className={styles.gwendyWendell}>
        <span className={styles.gwendyWrapper}>
          <span className={styles.gwendy}>Gwendy </span>
        </span>
        <span className={styles.wendellWrapper}>
          <span className={styles.wendell}>Wendell</span>
        </span>
      </a>
      <a href='/' className={styles.tcd}>
        <sup className={styles.the}>The</sup>
        <span className={styles.codingWrapper}><span className={styles.coding}>Coding</span> </span>
        <span className={styles.designerWrapper}><span className={styles.designer}>Designer</span></span>
      </a>
    </div>
  );
}
