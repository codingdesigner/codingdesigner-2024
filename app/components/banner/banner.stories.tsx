import React from 'react';
import type {Meta, StoryObj} from '@storybook/react';

import {Banner} from './banner';

const meta: Meta<typeof Banner> = {
  title: 'UI/Header/Banner',
  component: Banner,
};

export default meta;

type Story = StoryObj<typeof Banner>;

export const Alone: Story = {args: {}};
