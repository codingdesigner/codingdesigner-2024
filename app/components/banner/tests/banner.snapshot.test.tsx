import '@testing-library/jest-dom'
import {render } from '@testing-library/react'

import { Banner } from '../banner';

describe('<Banner />', () => {});


describe('<Banner />', () => {
  it('renders banner unchanged', () => {
    const component = render(<Banner />);
    expect(component).toMatchSnapshot()
  })
});
