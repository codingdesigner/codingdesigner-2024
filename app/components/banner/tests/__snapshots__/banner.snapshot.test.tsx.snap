// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> renders banner unchanged 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="banner"
        role="banner"
      >
        <a
          class="gwendyWendell"
          href="/"
        >
          <span
            class="gwendyWrapper"
          >
            <span
              class="gwendy"
            >
              Gwendy 
            </span>
          </span>
          <span
            class="wendellWrapper"
          >
            <span
              class="wendell"
            >
              Wendell
            </span>
          </span>
        </a>
        <a
          class="tcd"
          href="/"
        >
          <sup
            class="the"
          >
            The
          </sup>
          <span
            class="codingWrapper"
          >
            <span
              class="coding"
            >
              Coding
            </span>
             
          </span>
          <span
            class="designerWrapper"
          >
            <span
              class="designer"
            >
              Designer
            </span>
          </span>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="banner"
      role="banner"
    >
      <a
        class="gwendyWendell"
        href="/"
      >
        <span
          class="gwendyWrapper"
        >
          <span
            class="gwendy"
          >
            Gwendy 
          </span>
        </span>
        <span
          class="wendellWrapper"
        >
          <span
            class="wendell"
          >
            Wendell
          </span>
        </span>
      </a>
      <a
        class="tcd"
        href="/"
      >
        <sup
          class="the"
        >
          The
        </sup>
        <span
          class="codingWrapper"
        >
          <span
            class="coding"
          >
            Coding
          </span>
           
        </span>
        <span
          class="designerWrapper"
        >
          <span
            class="designer"
          >
            Designer
          </span>
        </span>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
